adapter/internal/discovery/xds/server.go

```diff
@@ -271,8 +271,13 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
 		apiEnvProps = apiEnvPropsV
 	}
 
+	mgwSwagger, err = operator.GetMgwSwaggerFromAPIYaml(apiProject.APIYaml, apiProject.APIType)
```

Comment:
line 284 getting security definitions from api.yaml, shouldn't it be applicable to both ws and rest apis? if so shall we move that also to this GetMgwSwaggerFromAPIYaml method

Reply:
API key option, X-wso2-auth-header options are not available for Websocket APIs. Hence, it would be okay to keep as it is I guess.